crates/rpc/rpc-builder/src/metrics.rs

```diff
@@ -1,87 +1,142 @@
 use jsonrpsee::{
     helpers::MethodResponseResult,
     server::logger::{HttpRequest, Logger, MethodKind, Params, TransportProtocol},
+    RpcModule,
 };
 use reth_metrics::{
     metrics::{Counter, Histogram},
     Metrics,
 };
-use std::{net::SocketAddr, time::Instant};
+use std::{collections::HashMap, net::SocketAddr, ops::Deref, time::Instant};
 
-/// Metrics for the rpc server
-#[derive(Metrics, Clone)]
-#[metrics(scope = "rpc_server")]
+/// Metrics for the RPC server
+#[derive(Default, Clone)]
 pub(crate) struct RpcServerMetrics {
-    /// The number of calls started
-    calls_started: Counter,
-    /// The number of successful calls
-    successful_calls: Counter,
-    /// The number of failed calls
-    failed_calls: Counter,
+    /// Connection metrics per transport type
+    connection_metrics: ConnectionMetrics,
+
+    /// Call metrics per RPC method
+    call_metrics: HashMap<&'static str, RpcServerCallMetrics>,
```

Comment:
`connection_metrics` should be fine, as it already has all metrics as `Arc` under the hood. Agree for `call_metrics` to avoid the cloning of hashmap, fixed.